### PR TITLE
[ADF-4933] Adding keydown event on data row to simulate double click when pressing 'enter' key

### PR DESCRIPTION
**Pull request by ursedaniel**
…eopening the file after closing ussing only keyboard (opening and closing the file by enter)

**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://issues.alfresco.com/jira/browse/ADF-4933
Currently if operating only with keyboard, when opening a file with 'enter' key it is displayed in the viewer component. If I try to close the viewer with the 'enter' key again, the event triggers again and reopens the file over and over again because the keyup event is not cancellable.

**What is the new behaviour?**

Now the file opens and closes correctly using the 'enter' key.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/datatable/components/datatable/datatable.component.ts [hunk 1]
```diff
@@ -474,8 +474,8 @@ export class DataTableComponent implements AfterContentInit, OnChanges, DoCheck,
         this.clickObserver.next(dataRowEvent);
     }
 
-    onRowKeyUp(row: DataRow, e: KeyboardEvent) {
-        const event = new CustomEvent('row-keyup', {
+    onRowKeyDown(row: DataRow, e: KeyboardEvent) {
+        const event = new CustomEvent('row-keydown', {
```
Comment (DenysVuika) [inline lib/core/datatable/components/datatable/datatable.component.ts]: this is a breaking change

File: lib/core/datatable/components/datatable/datatable.component.ts [hunk 2]
```diff
@@ -474,6 +474,12 @@ export class DataTableComponent implements AfterContentInit, OnChanges, DoCheck,
         this.clickObserver.next(dataRowEvent);
     }
 
+    onRowEnterKeyDown(row: DataRow, e: KeyboardEvent) {
+        if (e.key === 'Enter') {
+            this.onKeyboardNavigate(row, e);
+        }
+    }
+
     onRowKeyUp(row: DataRow, e: KeyboardEvent) {
```
Comment (eromano) [inline lib/core/datatable/components/datatable/datatable.component.ts]: avoid use os single char name variable prefer event over e that help the redability of the code when the variable is used far from the type

File: lib/core/datatable/components/datatable/datatable.component.ts [hunk 3]
```diff
@@ -474,6 +474,12 @@ export class DataTableComponent implements AfterContentInit, OnChanges, DoCheck,
         this.clickObserver.next(dataRowEvent);
     }
 
+    onRowEnterKeyDown(row: DataRow, e: KeyboardEvent) {
```
Comment (eromano) [inline lib/core/datatable/components/datatable/datatable.component.ts]: avoid use os single char name variable prefer event over e that help the redability of the code when the variable is used far from the type

**Comment by eromano** [comment]
I am going to merge this PR  also if there is a PS-Cloud failing I will ask someone to investigate that failing that doesn't look related at all with this PR

**Pull request merged**